chirp/drivers/retevis_ha1g.py

```diff
@@ -0,0 +1,2111 @@
+            write_items(self, self.pipe)
+        else:
+            raise errors.RadioError(error_map.get(
+                handshake_result, "Unknown error communicating with radio"))
+    except errors.RadioError:
+        raise
+    except Exception as e:
+        LOG.error(f"upload error: {e}")
+        raise errors.RadioError("Unknown error communicating with radio")
+    finally:
+        exit_programming_mode(self)
+
+
+def handshake(self, serial):
+    databytes = b""
+    for num in range(15):
+        flag, databytes = handle_connect_ver(
+            get_handshake_bytes(self.MODEL + " "), serial, self.read_page_len)
+        time.sleep(0.05)
+        if flag:
+            break
+    return handshake_handle_connect_event(self, databytes)
+
+
+def read_items(self, serial):
+    all_bytes = bytearray(self._memsize)
+    status = chirp_common.Status()
+    status.msg = "Cloning from radio"
+    status.cur = 0
+    status.max = self._memsize
+    for item in MemoryRegions:
+        try:
+            item_bytes = get_read_current_page_bytes(
+                self, item.value, serial, status)
+            if item_bytes:
+                all_bytes = read_item_handle_connect_event(
+                    all_bytes, item_bytes, item)
+        except Exception as e:
+            LOG.error(
+                f"read item_data error: {item.name} error_msg: {e}")
+            continue
+    return all_bytes
+
+
+def write_items(self, serial):
+    status = chirp_common.Status()
+    status.msg = "Uploading to radio"
+    status.cur = 0
+    status.max = self._memsize
+    data_bytes = self.get_mmap()
+    for item in MemoryRegions:
+        if (item in [MemoryRegions.radioHead, MemoryRegions.radioInfo,
+                     MemoryRegions.radioVer, MemoryRegions.zoneData]):
+            continue
+        item_bytes = get_write_item_bytes(data_bytes, item)
+        write_item_current_page_bytes(
+            self, serial, item_bytes, item.value, status)
+
+
+def handshake_handle_connect_event(self, dataByte: bytes):
+    if dataByte[14:16] == b"\x00\x01":
+        if dataByte[20] != 1:
+            model_str = self.MODEL
+            radioType = dataByte[20: 20 + len(model_str)]
+            if radioType == model_str.encode("ascii"):
+                return HandshakeStatuses.Normal
+            else:
+                return HandshakeStatuses.RadioWrong
+        else:
+            return HandshakeStatuses.PwdWrong
+    else:
+        return HandshakeStatuses.Wrong
+
+
+def handle_connect_ver(
+    send_byte: bytes, serial_conn, chunk_size=1050, max_retries=5
+) -> tuple[bool, bytes]:
+    serial_conn.write(send_byte)
+    new_bytes = serial_conn.read(chunk_size)
+    if rdtp_page_data_crc16_ver(new_bytes):
+        return True, new_bytes
+    return False, new_bytes
+
+
+def rdtp_page_data_crc16_ver(current_Page_Byte: bytes, chunk_size=1024):
+    byteLen = len(current_Page_Byte)
+    if byteLen <= 13:
+        return False
```

Comment:
Please don't do this. If this function checks the CRC for a result, name it accordingly and make it do the thing. As best I can tell, this will return True if the CRC matches, or False if the CRC does not match *OR* False if the amount of data is too small to be CRC'd. That's a very bad behavior for a function to have..